components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1269,7 +1270,139 @@ where
             SignificantMsg::RaftlogFetched { context, res } => {
                 self.on_raft_log_fetched(context, res);
             }
+            SignificantMsg::EnterForceLeaderState { failed_stores } => {
+                self.on_enter_pre_force_leader(failed_stores)
+            }
+            SignificantMsg::ExitForceLeaderState => self.on_exit_force_leader(),
+        }
+    }
+
+    fn on_enter_pre_force_leader(&mut self, failed_stores: HashSet<u64>) {
+        if self.fsm.peer.force_leader.is_some() {
```

Comment:
Should check whether `expected_alive_voter` changes before ignoring.

Reply:
It's possible. For example, the connection is lost when PD asks it to exit force_leader mode.